lib/src/components/NxGlobalSidebar/__tests__/NxGlobalSidebarNavigationLink.test.tsx

```diff
@@ -20,33 +18,80 @@ describe('NxGlobalSidebarNavigationLink', function() {
         text: 'textLink',
         href: '#someurl'
       },
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxGlobalSidebarNavigationLink, minimalProps);
+      quickRender = rtlRender<Props>(NxGlobalSidebarNavigationLink, minimalProps);
+
+  describe('tooltips', function () {
+    it('renders a tooltip on hover if link text overflow occurs', async function() {
+      // Supress overflow tooltip warnings
+      jest.spyOn(console, 'warn').mockImplementation(() => {});
+
+      // Mock layout methods imported from NxOverflowTooltip RTL test.
+      jest.spyOn(Element.prototype, 'getBoundingClientRect').mockReturnValue({
+        x: 0,
+        y: 0,
+        width: 1,
+        height: 1,
+        top: 0,
+        right: 1,
+        bottom: 1,
+        left: 0
+      } as DOMRect);
+
+      // mock that text extends 1px farther than container
+      jest.spyOn(Range.prototype, 'getBoundingClientRect').mockReturnValue({
+        x: 0,
+        y: 0,
+        width: 2,
+        height: 1,
+        top: 0,
+        right: 2,
+        bottom: 1,
+        left: 0
+      } as DOMRect);
+
+      const view = quickRender(),
+          link = view.getByRole('link'),
+          user = userEvent.setup();
+
+      await user.hover(link);
+
+      const tooltip = await screen.findByRole('tooltip');
+
+      expect(tooltip).toBeInTheDocument();
+      expect(tooltip).toHaveTextContent('textLink');
+    });
+
+    it('does not render a tooltip on hover if link text overflow doesn\'t occur', async function() {
+      const view = quickRender(),
+          link = view.getByRole('link'),
+          user = userEvent.setup();
+
+      await user.hover(link);
+
+      expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+    });
 
-  it('renders an NxOverflowTooltip as container', function () {
-    expect(getShallowComponent()).toMatchSelector(NxOverflowTooltip);
   });
 
   it('renders an <a> with the passed href and classes', function() {
-    const link = getShallowComponent({ className: 'extra-class' }).find('a');
-    expect(link).toMatchSelector('a.nx-global-sidebar__navigation-link.nx-text-link');
-    expect(link).toMatchSelector('.extra-class');
-    expect(link).toHaveProp('href', '#someurl');
+    const link = quickRender({ className: 'extra-class' }).getByRole('link');
+    expect(link).toHaveClass('extra-class');
```

Comment:
maybe test classes as we have in other tests with the comparison of an element with default classes vs a custom class name

Reply:
> maybe test classes as we have in other tests with the comparison of an element with default classes vs a custom class name

Added here: 56471ee4b334cff22b1734ff484818241c41cd87